spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -155,23 +405,463 @@ class NwbSortingExtractor(se.SortingExtractor):
+        print(self.clear_units_property.__doc__)
+
+    def get_nspikes(self):
+        """Returns list with the number of spikes for each unit."""
+        # todo: there is a way to do this without reading all of the data if you use spike_times_index
+        assert HAVE_NWB, "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
+        with NWBHDF5IO(self._path, 'r+') as io:
+            nwbfile = io.read()
+            nSpikes = [len(spkt) for spkt in nwbfile.units['spike_times'][:]]
+        return nSpikes
+
+    def get_unit_spike_features(self, unit_id, feature_name, start_frame=0, end_frame=np.Inf):
+        '''This function extracts the specified spike features from the specified unit.
+        It will return spike features from within three ranges:
+
+            [start_frame, t_start+1, ..., end_frame-1]
+            [start_frame, start_frame+1, ..., final_unit_spike_frame - 1]
+            [0, 1, ..., end_frame-1]
+            [0, 1, ..., final_unit_spike_frame - 1]
+
+        if both start_frame and end_frame are given, if only start_frame is
+        given, if only end_frame is given, or if neither start_frame or end_frame
+        are given, respectively. Spike features are returned in the form of an
+        array_like of spike features. In this implementation, start_frame is inclusive
+        and end_frame is exclusive conforming to numpy standards.
+
+        Parameters
+        ----------
+        unit_id: int
+            The id that specifies a unit in the recording.
+        feature_name: string
+            The name of the feature to be returned.
+        start_frame: int
+            The frame above which a spike frame is returned  (inclusive).
+        end_frame: int
+            The frame below which a spike frame is returned  (exclusive).
+        Returns
+        ----------
+        spike_features: numpy.ndarray
+            An array containing all the features for each spike in the
+            specified unit given the range of start and end frames.
+        '''
+        assert HAVE_NWB, "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
+        if not isinstance(feature_name, str):
+            raise Exception("'feature_name' must be a string")
+        full_feat_name = 'spike_feature_' + feature_name
+        if full_feat_name not in self.get_shared_unit_spike_feature_names():
+            raise Exception(full_feat_name + " not present in NWB file")
+
+        with NWBHDF5IO(self._path, 'r') as io:
+            nwbfile = io.read()
+            units = nwbfile.units
+            # chosen unit and interval
+            feat_vals = np.array(units[full_feat_name][units.id.index(unit_id)][:])
+            times = units['spike_times'][units.id.index(unit_id)][:]
+            # spike times are measured in samples
+            frames = self.time_to_frame(times)
+            mask = (frames > start_frame) & (frames < end_frame)
```

Comment:
```suggestion
            mask = (frames >= start_frame) & (frames < end_frame)
```
since doc said "inclusive"